File: examen.js
function gets()
{
    var reader = new java.io.InputStreamReader(java.lang.System.in); 
    var buffer = new java.io.BufferedReader(reader); 
    var line = buffer.readLine();
    return line;
}

function puts() {
    for( var i = 0; i < arguments.length; i++ ) {
       var value = arguments[i];
       java.lang.System.out.print( value );
    }
}

function media(set) {
    var suma = 0;
    for (let i = 0; i < set.length; i++) {
        suma = parseInt(set[i]) + suma;
    }
    var result = suma/set.length;
    return result;
}


function varianza (set, average) {
    var diferencia = 0;
    var diferenciaSq = 0;
    var numero = 0;
    for (let iteration = 0; iteration < set.length; iteration++) {
        diferencia = parseInt(average) - parseInt(set[iteration]);
        diferenciaSq = (diferencia*diferencia);
        numero = (numero + diferenciaSq);
    }
    var varianzaAlt = numero/set.length;
    return varianzaAlt;
}

function modaDeConjunto(set) {
    var iteration = 0;
    var iteration_ = 0;
    var repetition = 0;
    var moda = 0;
    var count = 0;
    repetition = -1;
    for(iteration = 0; iteration < set.length; iteration++){
        count = 1;
        for(iteration_ = iteration+1; iteration_ < set.length; iteration_++){
            if(parseInt(set[iteration_]) == parseInt(set[iteration])){
                count++;
                if(count > repetition){
                    repetition = count;
                    moda = parseInt(set[iteration_]);
                }
            }
        }
    }
    return moda;
}




puts("Número de alumnos: ");
var alumnCount = parseInt(gets());
var alumnGrades = new Array(alumnCount);
for (let iteration = 0; iteration < alumnGrades.length; iteration++) {
    var visIndex = iteration + 1;
    puts("Ingrese la calificación del alumno ", visIndex.toString(), ": ");
    alumnGrades[iteration] = gets();
}
const average = media(alumnGrades);
const varianz = varianza(alumnGrades, average).toFixed(6);
const desviac = Math.sqrt(varianz).toFixed(6);
const moda    = modaDeConjunto(alumnGrades);
puts("\n", "Media aritmética   : ", average.toString());
puts("\n", "Varianza           : ", varianz.toString());
puts("\n", "Desviación estándar: ", desviac.toString());
puts("\n", "Moda               : ", moda.toString(),"\n");

File: examen.c
#include<stdio.h>
#include <math.h>

float sumaDeConjunto (int set[], int cardinality) {
    int suma = 0;
    for (int iteration = 0; iteration < cardinality; iteration++) {
        suma = suma + set[iteration];
    }
    return suma;
}

float varianza (int set[], int cardinality, float average) {
    float diferencia;
    float diferenciaSq;
    float numero;
    for (int iteration = 0; iteration < cardinality; iteration++) {
        diferencia = average - set[iteration];
        diferenciaSq = (diferencia*diferencia);
        numero = (numero + diferenciaSq);
    }
    float varianzaAlt = (numero/cardinality);
    return varianzaAlt;
}

int modaDeConjunto(int set[], int cardinality) {
    int iteration, iteration_, repetition, moda, count;
    repetition = -1;
    for(iteration = 0; iteration< cardinality; iteration++){
        count = 1;
        for(iteration_ = iteration+1; iteration_ < cardinality; iteration_++){
            if(set[iteration_] == set[iteration]){
                count++;
                if(count > repetition){
                    repetition = count;
                    moda = set[iteration_];
                }
            }
        }
    }
    return moda;
}

int main()
{
    int alumnCount;
    printf("Número de alumnos: ");
    scanf("%d", &alumnCount);
    int alumnGrades[alumnCount];
    for (int iteration = 0; iteration < alumnCount; iteration++) {
        printf("Calificación del alumno %d: ", iteration+1);
        scanf("%d", &alumnGrades[iteration]);
    }

    float mediaAritmetica = (sumaDeConjunto(alumnGrades, alumnCount)/alumnCount);
    printf("\nMedia aritmética: %g",mediaAritmetica);
    float Varianza = varianza(alumnGrades, alumnCount, mediaAritmetica);
    printf("\nVarianza: %g\n", Varianza);
    float desviacionEstandar = sqrt(Varianza);
    printf("Desviación estándar: %g\n", desviacionEstandar);
    int moda = modaDeConjunto(alumnGrades, alumnCount);
    printf("Moda: %d\n", moda);
}